airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -130,6 +130,13 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                     stream_is_available,
                     reason,
                 ) = stream.availability_strategy.check_availability_and_parsability(stream, logger, self)
+            except EncodingError as encoding_error:
```

Comment:
we shouldn't handle EncodingErrors differently here.

My understanding from your experiment:
- errors only show properly if we set the `message` field and raise the exception

This means what we're doing today with the tracebacks is wrong.

Instead, we could:
- create an error message from the errors list. if there's more than one, maybe it should only show the streams that couldn't be synced.
- we can still concatenate the traceback as the TracedException's, internal message for troubleshooting purposes

The problem with the current approach is we're only fixing the rendering for encoding errors, not for others

Reply:
secondary problem is that encoding errors are now the only exception that result in a complete failure without trying subsequent streams. that's inconsistent with other types

Reply:
I think your point is right. I feel I still wouldn't like the result because, in the end, the user is going to need to read a huge traceback. I believe the right approach is raise each known exception as I did with the EncodingError, to be able to provide a meaningful error, well formatted in the UI. In the case of many failed streams, we should then collect the right error messages for at least show one meaningful error in the UI, and the rest can be in the traceback. However, with the current structure of the code, not even one good message can be saved, right? The information needed to create a good error message is lost.